orc8r/tools/fab/dev_utils.py

```diff
@@ -405,3 +407,43 @@ def cloud_delete(
             'Delete Request failed: \n%s \nReceived a %d response: %s\nFAILED!' %
             (resp.url, resp.status_code, resp.text),
         )
+
+
+def local_command_with_repetition(command, timeout=5):
+    """
+    run command on local machine using fabric.api.local. Repeats on error
+    Args:
+        command: command to issue
+        timeout: time to execute the command while it fails
+    """
+    _command_with_repetition(local, command, timeout)
+
+
+def run_fab_command_with_repetition(command, timeout=5):
+    """
+    run command on remote machine using fabric.api.run. Repeats on error
+    Args:
+        command: command to run
+        timeout: time to execute the command while it fails
+    """
+    _command_with_repetition(run, command, timeout)
+
+
+def _command_with_repetition(func, command, timeout=5):
```

Comment:
Nit: "timeout" gives the connotation it's time-based. Maybe something like "max retries"?

Reply:
In fact it is a time out.  We are not worried for the number of attempts (which happen every second)

The reason for that is on those integ test normally the system is not ready by the time the test starts. Adding this allow us to have some control on how long will the test retry.  This command does not pretend to be exact (if it takes more tjhan 5 seconds is fine, it will not take less in case the system is down)

In other words, this just mimics what a person would do while waiting for the service to start (similar to running a watch command)

If i am able to add this tests on github actions , I will call that command with 60 to make sure I give the system plenty of time to start and bootstrap

Reply:
Hm if I'm reading correctly, it looks like there's no timing functionality specifically in this function -- looks like it increments a count after every attempt, and after a max number of retires gives up

To me, timeout means "try for at most x seconds", while max tries means "try at most this number of times". Not a huge deal, and maybe I'm reading it wrong 🙂

Reply:
check line 459

Reply:
Right right. I guess the consideration is still: consider if "func" takes e.g. 10 minutes to run -- then the total runtime would be `timeout*10m1s`, rather than `timeout*1s`. Which makes it feel like a max_retries rather than timeout

But again, just a nit, up to you!

Reply:
The commands used by the fab command here are normally limited in time. But since the code is in a shared library it may be wise to add an upper bound. Will change some things so we can make the function more generic then